src/ert/gui/ertwidgets/ensembleselector.py

```diff
@@ -69,9 +72,26 @@ def populate(self) -> None:
             self.setEnabled(True)
 
         for ensemble in self._ensemble_list():
+            model = cast(QStandardItemModel, self.model())
             self.addItem(
                 f"{ensemble.experiment.name} : {ensemble.name}", userData=ensemble
             )
+            if (
+                self._show_only_with_valid_experiment
```

Comment:
Do you really need to check for `_show_only_with_valid_experiment` here?
Does it not make sense to just do this for all experiments that are not valid?